core/block_svg.ts

```diff
@@ -1161,7 +1161,10 @@ export class BlockSvg
       .getRenderer()
       .getConstants()
       .getBlockStyle(blockStyleName);
-    this.styleName_ = blockStyleName;
```

Comment:
Can you re-add this after your check so that the style name is still being cached? I don't think your logic quite works without this :/